src/main/scala/de/heikoseeberger/sbtheader/HeaderCreator.scala

```diff
@@ -64,20 +66,22 @@ final class HeaderCreator private (fileType: FileType,
     }
 
   private val headerNewLine =
-    headerText match {
+    hText match {
       case crlf(_) => "\r\n"
       case cr(_)   => "\r"
       case _       => "\n"
     }
 
-  private val newHeaderText = headerText.replace(headerNewLine, fileNewLine)
+  private val newHeaderText = hText.replace(headerNewLine, fileNewLine)
+  private val hTextNl       = hText + newLine // the regex always checks for a new line at the end of a header text
+  private val nl            = (if (headerEmptyLine) newLine else "")
```

Comment:
this is a bad name because is it not always a new line. It is either a new line or an empty string. Since the new line at the end of the comment block is optional, I think this should be of type `Option[String]` instead of using an empty string a marker for "no new line"